diff -up openmw-openmw-0.48.0/apps/openmw/mwstate/charactermanager.hpp.omv~ openmw-openmw-0.48.0/apps/openmw/mwstate/charactermanager.hpp
--- openmw-openmw-0.48.0/apps/openmw/mwstate/charactermanager.hpp.omv~	2024-03-28 02:44:52.095980408 +0100
+++ openmw-openmw-0.48.0/apps/openmw/mwstate/charactermanager.hpp	2024-03-28 02:45:00.467167247 +0100
@@ -1,6 +1,7 @@
 #ifndef GAME_STATE_CHARACTERMANAGER_H
 #define GAME_STATE_CHARACTERMANAGER_H
 
+#include <list>
 #include <boost/filesystem/path.hpp>
 
 #include "character.hpp"
