diff -uraN openmw-openmw-0.50.0/components/sceneutil/texturetype.hpp omv-openmw-openmw-0.50.0/components/sceneutil/texturetype.hpp
--- openmw-openmw-0.50.0/components/sceneutil/texturetype.hpp	2025-10-28 18:14:40.000000000 +0100
+++ omv-openmw-openmw-0.50.0/components/sceneutil/texturetype.hpp	2025-11-08 17:52:00.608597837 +0100
@@ -18,7 +18,7 @@
         {
         }
 
-        static const osg::StateAttribute::Type AttributeType = static_cast<osg::StateAttribute::Type>(69420);
+        static const osg::StateAttribute::Type AttributeType = static_cast<osg::StateAttribute::Type>(110);
         META_StateAttribute(SceneUtil, TextureType, AttributeType)
 
         bool isTextureAttribute() const override { return true; }
