diff -up openmw-openmw-0.47.0/apps/openmw/mwinput/controlswitch.hpp.omv~ openmw-openmw-0.47.0/apps/openmw/mwinput/controlswitch.hpp
--- openmw-openmw-0.47.0/apps/openmw/mwinput/controlswitch.hpp.omv~	2023-07-21 01:15:24.286206585 +0200
+++ openmw-openmw-0.47.0/apps/openmw/mwinput/controlswitch.hpp	2023-07-21 01:15:33.509279794 +0200
@@ -1,6 +1,7 @@
 #ifndef MWINPUT_CONTROLSWITCH_H
 #define MWINPUT_CONTROLSWITCH_H
 
+#include <cstdint>
 #include <map>
 #include <string>
 
diff -up openmw-openmw-0.47.0/components/misc/utf8stream.hpp.omv~ openmw-openmw-0.47.0/components/misc/utf8stream.hpp
--- openmw-openmw-0.47.0/components/misc/utf8stream.hpp.omv~	2023-07-21 01:11:34.784520184 +0200
+++ openmw-openmw-0.47.0/components/misc/utf8stream.hpp	2023-07-21 01:11:40.597559164 +0200
@@ -1,6 +1,7 @@
 #ifndef MISC_UTF8ITER_HPP
 #define MISC_UTF8ITER_HPP
 
+#include <cstdint>
 #include <cstring>
 #include <tuple>
 
